### DISCUSSION TITLE
How to check for translation errors?

### DISCUSSION THREAD

**Discussion by honeyhl** [ptitSeb/box86#572] [Q&A]
When I run wine with box86, the target program cannot exit. I found that wineserver was blocked in epoll_wait with gdb attach, and no other obvious errors were found. The latest code has been updated to solve this problem. If I encounter this problem next time, how can I locate this problem? What about types of problems? How to check for translation errors?

**Comment by ptitSeb** [comment]
There is no general rules, each problem needs it's own solution.
But, to be able to debug, you need to a "trace enabled" build, and you need to build Zydis disassembler.
When you have a working trace build, you launch stuff with `BOX86_TRACE=0-1 BOX86_TRACE_FILE=dump.txt BOX86_DYNAREC_DUMP=2` to have all x86->arm block dumpped to a file (dump.txt). Use `less -R dump.txt` to read the file, as it use some color in the output. Use `BOX86_DYNAREC_DUMP=1` to not use colored output.
Also, if debug wine stuff, you will have multiple process running, each will write some dump. To avoid all the process trying to write the same dump, use `BOX86_TRACE_FILE=$(pwd)/dump-%pid.txt` instead, to have each process write it's own unique file.
